src/plugins/intel_cpu/src/transformations/snippets/aarch64/pass/lowered/expressions/gemm_copy_b_buffer_expressions.cpp

```diff
@@ -0,0 +1,80 @@
+RepackedWeightsBufferExpression::RepackedWeightsBufferExpression(
+    const std::shared_ptr<ov::Node>& n,
+    const std::shared_ptr<snippets::IShapeInferSnippetsFactory>& factory)
+    : BufferExpression(n, factory) {}
+
+snippets::lowered::ExpressionPtr RepackedWeightsBufferExpression::clone() const {
+    return std::make_shared<RepackedWeightsBufferExpression>(*this);
+}
+
+void RepackedWeightsBufferExpression::validate() const {
+    BufferExpression::validate();
+    OPENVINO_ASSERT(get_input_count() == 1, "RepackedWeightsBufferExpression must have only one input");
+    const auto& parent_out = get_input_port_connector(0)->get_source();
+    OPENVINO_ASSERT(ov::is_type<ov::intel_cpu::aarch64::GemmCopyB>(parent_out.get_expr()->get_node()) &&
+                        parent_out.get_index() == 0,
+                    "RepackedWeightsBufferExpression expects GemmCopyB as parent expression");
+    OPENVINO_ASSERT(one_of(get_node()->get_input_element_type(0), ov::element::f32),
+                    "RepackedWeightsBufferExpression after GemmCopyB currently only support f32 data type on arm");
+}
+
+void RepackedWeightsBufferExpression::init_allocation_size(
+    [[maybe_unused]] const std::shared_ptr<snippets::lowered::LoopManager>& loop_manager,
+    [[maybe_unused]] size_t allocation_rank) {
+    const auto& parent_expr = get_input_port_connector(0)->get_source().get_expr();
+    const auto& in_subtensor = ov::snippets::utils::get_projected_subtensor(parent_expr->get_input_port(0));
+    OPENVINO_ASSERT(in_subtensor.size() >= 2 && allocation_rank >= 2, "GemmCopyB should has at least 2 rank tensor");
+    const auto& element_type = get_node()->get_input_element_type(0);
+    const size_t N = *in_subtensor.rbegin();
+    const size_t K = *++in_subtensor.rbegin();
+
+    const auto& consumers = get_output_port_connector(0)->get_consumers();
+    ExpressionPtr child_gemm_expr = nullptr;
+    // maybe connected to loopEnd besides gemm
+    for (const auto& consumer : consumers) {
+        if (ov::is_type<ov::intel_cpu::aarch64::GemmCPU>(consumer.get_expr()->get_node())) {
+            child_gemm_expr = consumer.get_expr();
+            break;
+        }
+    }
+    OPENVINO_ASSERT(child_gemm_expr, "RepackedWeightsBufferExpression must connect to gemm");
+    const auto& gemm_in_subtensor = ov::snippets::utils::get_projected_subtensor(child_gemm_expr->get_input_port(1));
+    const size_t n_block_size = *gemm_in_subtensor.rbegin();
+    if (snippets::utils::is_dynamic_value(N) || snippets::utils::is_dynamic_value(K) ||
+        snippets::utils::is_dynamic_value(n_block_size)) {
+        m_allocation_size = snippets::utils::get_dynamic_value<size_t>();
+        return;
+    }
+    size_t n_block_num = N / n_block_size;
+    size_t n_tail_size = N % n_block_size;
+    m_allocation_size = 0;
+    for (size_t i = 0; i < n_block_num; i++) {
+        m_allocation_size += kai_get_rhs_packed_size_rhs_pack_kxn_f32p8x1biasf32_f32_f32_neon(n_block_size, K);
```

Comment:
What's about `m_allocation_size = n_block_num * kai_get_rhs_packed_size_rhs_pack_kxn_f32p8x1biasf32_f32_f32_neon(n_block_size, K)`?

Or even `kai_get_rhs_packed_size_rhs_pack_kxn_f32p8x1biasf32_f32_f32_neon(N, K)`? (I see that the plugin [passes the both dimensions](https://github.com/chenhu-wang/openvino/blob/d1c65e2d8ab9713f781506b87e263ee35d67f1a8/src/plugins/intel_cpu/src/nodes/executors/kleidiai/kleidiai_mm.cpp#L92)) Or might there be some incorrect calculations?